packages/react-native-windows-init/src/Cli.ts

```diff
@@ -519,10 +625,20 @@ function isProjectUsingYarn(cwd: string): boolean {
       useHermes: argv.useHermes,
       nuGetTestVersion: argv.nuGetTestVersion,
       nuGetTestFeed: argv.nuGetTestFeed,
+      noTelemetry: argv.noTelemetry,
     });
+    return setExit(ExitCode.SUCCESS);
   } catch (error) {
-    console.error(chalk.red(error.message));
-    console.error(error);
-    process.exit(EXITCODE_UNKNOWN_ERROR);
+    const exitCode =
+      error instanceof UserError
+        ? (error as UserError).exitCode
+        : ExitCode.UNKNOWN_ERROR;
+    if (exitCode !== ExitCode.SUCCESS) {
+      console.error(chalk.red(error.message));
+      console.error(error);
+    }
+    setExit(exitCode, error.message);
+  } finally {
+    telClient.flush();
```

Comment:
nit: In other telemetry pipelines I have seen that it might appear that apps 'hang' if uploading the telemetry takes too long... Not sure if the JS version of app insights uses the offline sync mechanism that windows uses, or if it will wait for HTTP response in the main execution path....

Reply:
So perhaps a log timer that if it takes more than 200ms prints 'sending telemetry...'

Reply:
I don't see a way in AI to enable this but we can follow up offline if need be :)